### DISCUSSION TITLE
So nice to find this, I was almost thinking about using Clasp.

### DISCUSSION THREAD

**Discussion by alex-ca1123** [jank-lang/jank#65] [Issue Triage] [answered]
but java's type system is much more straightforward than cpp, I can imagine that cpp interop would not be easy. Thank god we have LLVM now. Otherwise the alignment between JVM, x86, Arm and other machine codes would be impossible without a big commercial engineering team.

**Comment by jeaye** [comment, marked as answer]
> I was almost thinking about using Clasp.

If you need something today, clasp may be your best choice. jank isn't ready for use just yet, but I aim to have a rough-but-workable version distributed to folks in a few months.

> java's type system is much more straightforward than cpp, I can imagine that cpp interop would not be easy.

You're right about this. With the JVM, everything is an Object, so any of Clojure's collections can old any value you can throw at it. C++ has no notion of this, so we need to explicitly type-erase things, which involves an allocation and extra code when you put it in and take it out of the jank system.

> Thank god we have LLVM now.

The main work here started at CERN, when they built Cling (a JIT C++ compiler). Right now, jank uses Cling behind the scenes, to make all of the REPL JIT goodness come to life. But all of this is getting upstreamed into LLVM and the JIT ecosystem in LLVM is growing very quickly (a PR for JIT reoptimization just landed in LLVM in the past week!). There are still some blocker bugs on the LLVM side which prevent jank from switching away from Cling to use the new LLVM version, but it'll be a big win once we can.

If you do have C++ and/or Clojure experience and you're looking to lend a hand, let me know! Otherwise, hang tight and watch for my monthly blog posts. :slightly_smiling_face:

**Reply by pauldorman** [reply]
Came here from the Clojurists Together funding announcement to say: might not be using Jank yet, but I will be! Brace yourself for the impending stampede!!! Bootstraping a language is a bitch, but when you cross that invisible line it will be glorious!

**Reply by jeaye** [reply]
> Came here from the Clojurists Together funding announcement to say: might not be using Jank yet, but I will be! Brace yourself for the impending stampede!!! Bootstraping a language is a bitch, but when you cross that invisible line it will be glorious!

Hey! Welcome to the community. :purple_heart: 

jank is closer and closer to its initial release. Right now, Clojure syntax parity is great, and clojure.core support is pretty good, but the tooling is all unusable. I'm working on getting AOT compilation and nREPL support done and that, I think, will put us in a position to where people can start actually building some things with jank.

If you're interested in helping out, let me know! Plenty of work to be done in C++, Clojure, and other tasks such as packaging and distribution.
